decisions/websre/0005-web-sre-kubernetes-secrets-pattern.md

```diff
@@ -0,0 +1,40 @@
+
+There are currently a few ways the secrets can be written to the cloud secret store. We have some terraform that writes the secrets directly to the secrets stores, but most secrets are populated manually by operators.  We expect both methods to be used going forward.  In general, we expect SREs/Operators with elevated privileges to manage the secrets.  This ADR intentionally doesn't include scope for devs to write new secrets, or update existing ones.  That problem is as of today, unsolved.
+
+The default assumption for storage patterns in aws secrets store is that you follow the pattern: `<group>/<app>/<environment>/<secretname>` for example `mozmeao/snippets/stage/database_url`  This seems to make writing aws iam access stanzas easy.  Not needing to be rewritten with each new secret added.  But, enough separation to prevent us from accidentally crossing wires.
+
+We are choosing to not use a few other types of secret storage mechanisms, including (but possibly not limited to):
+* git-crypt secret repos + sops, to k8s secrets directly.
+* Sealed secrets (which uses the kubernetes certs as an asymmetric key for local storage, and a crd in the cluster to present the secrets as env variables to the app)
+* SSM Parameter Store (lambdas/codebuilds use this)
+* no new systems to be built, for example hashicorp vault
+
+We expect CI/CD pipelines to use their current secret storage, and not try to connect them to the secret store.  The one exception is the cloud native pipeline solutions (e.g AWS CodeBuild), which should use the secrets manager if possible.  We may end up duplicating secrets, but in general we should try to keep application secrets and pipeline secrets separate, with pipeline secrets being manually added in the right context.
+
+## Consequences
+
+
+We will be migrating secrets, which is a reasonably risky process for many applications.
+We will have fewer methods for managing secrets, which ought to make it easier to rotate secrets, keep them up to date, etc.
+We are gatekeepers for adding new secrets, which may increase unplanned work for our team.
+If we were to lose the secrets manager completely, rebuilding would be a long and arduous process.
+Updating secrets will include manual steps, like rotating pods, to use the new secret.
```

Comment:
This is already the case today for the SSL monitoring pods, so I had assumed in general that we would need to rotate pods in _all_ instances where we update credentials, which negates this as a consequence of our decision here. Am I mistaken, and there are places where we store secrets in ways other than aws/gcp secrets manager, that somehow rotate automatically?

Reply:
I don't think any of our existing methods automatically rotate secrets today.  I've used sidecars in the past that did do the auto-update.  I suppose this is a consequence of rejecting some theoretical new secret management method, but point taken that none of the other methods do this today.